src/state/slices/common-selectors.ts

```diff
@@ -200,6 +200,37 @@ export const selectAssetsSortedByMarketCapUserCurrencyBalanceAndName =
     },
   )
 
+export const selectAssetsSortedByMarketCapUserCurrencyBalanceCryptoBalanceAndName =
+  createDeepEqualOutputSelector(
+    selectAssets,
+    selectPortfolioAssetBalancesBaseUnit,
+    selectPortfolioUserCurrencyBalances,
+    selectMarketDataUsd,
+    (assets, portfolioCryptoBaseUnitBalances, portfolioUserCurrencyBalances, marketDataUsd) => {
+      const getAssetCryptoBaseUnitBalance = (asset: Asset) =>
+        bnOrZero(portfolioCryptoBaseUnitBalances[asset.assetId]).toNumber()
+      const getAssetUserCurrencyBalance = (asset: Asset) =>
+        bnOrZero(portfolioUserCurrencyBalances[asset.assetId]).toNumber()
+
+      // This looks weird but isn't - looks like we could use the sorted selectAssetsByMarketCap instead of selectAssets
+      // but we actually can't - this would rug the quadruple-sorting
+      const getAssetMarketCap = (asset: Asset) =>
+        bnOrZero(marketDataUsd[asset.assetId]?.marketCap).toNumber()
+      const getAssetName = (asset: Asset) => asset.name
+
+      return orderBy(
+        Object.values(assets).filter(isSome),
+        [
+          getAssetUserCurrencyBalance,
+          getAssetMarketCap,
+          getAssetCryptoBaseUnitBalance,
```

Comment:
blocking: we can't use base units here, different assets will yield different base units. e.g holding 1 ETH will yield a balance  of 1000000000000000000 wei while holding 1 BTC will hold a balance of 100000000 sats

Reply:
Nice catch 🐐 

[7a160b8](https://github.com/shapeshift/web/pull/8175/commits/7a160b87ffecbd5c7ec949303dcfc169794d066a)